File: Text Adventure/Text Adventure/Room.cs
using System;
using System.Collections.Generic;

namespace TextAdventure
{
    public class Room : IInspectable
    {
        public string Name { get; }
        public string Description { get; }
        public Dictionary<string, Room> Exits { get; }
        public List<IInspectable> Inspectables { get; } = new List<IInspectable>();
        public bool IsLocked { get; private set; }

        public Room(string name, string description, bool isLocked = true)
        {
            Name = name;
            Description = description;
            Exits = new Dictionary<string, Room>();
            IsLocked = isLocked;
        }

        public void AddExit(string direction, Room room)
        {
            Exits[direction] = room;
        }

        public void AddInspectable(IInspectable inspectable)
        {
            Inspectables.Add(inspectable);
        }

        public void Unlock()
        {
            IsLocked = false;
            Console.WriteLine($"{Name} has been unlocked!");
        }
        public void LoadRoomInventory(List<Item> items)
        {
         Inspectables.Clear();
         foreach (var item in items)
         {
              AddInspectable(item);
         }
        }


        public void Inspect()
        {
            Console.WriteLine($"{Name} - {Description}");

            if (Inspectables.Count > 0)
            {
                Console.WriteLine("You see:");
                foreach (var item in Inspectables)
                {
                    Console.WriteLine($"- {item.GetName()}");
                }
            }

            if (Exits.Count > 0)
            {
                Console.WriteLine("Exits:");
                foreach (var exit in Exits.Keys)
                {
                    Console.WriteLine($"- {exit}");
                }
            }
        }

        public string GetName()
        {
            return Name;
        }
    }
}

File: Text Adventure C++/Text Adventure C++/Room.cpp
#include "Room.hpp"
#include <iostream>
using namespace std;

Room::Room(const string& name, const string& description)
    : name(name), description(description) {}

void Room::AddExit(const string& direction, Room* room) {
    exits[direction] = room;
}

void Room::AddInspectable(const string& itemName) {
    inspectables.push_back(itemName);
}

void Room::Inspect() const {
    cout << "You are in " << name << ". " << description << "\n";

    if (!inspectables.empty()) {
        cout << "You see: ";
        for (const auto& item : inspectables) {
            cout << item << " ";
        }
        cout << "\n";
    }

    if (!exits.empty()) {
        cout << "Exits: ";
        for (const auto& exit : exits) {
        cout << exit.first << " ";
        }
        cout << "\n";
    }
}

Room* Room::GetExit(const string& direction) const {
    auto it = exits.find(direction); //used auto like var in c#
    if (it != exits.end()) {
        return it->second;
    }
    return nullptr; //Return null pointer if no exit is found
}

const vector<string>& Room::GetInspectables() const {
    return inspectables; 
}

